File: EX03_03/Python/BinaryRepPart2.py
# Yuriy Fedas - Data Representation Example - Part 2
def check(num):
    while (num != 0):
        if ((num & 3) == 3):
            return True
        else:
            num = num >> 1
    # Default case when num = 0 and no consecutive bits are found
    return False

def decrement(num):
    # less than the number entered
    num -= 1
    count = 0
    while (num != 0):
        if (check(num)):
            count += 1
        num -= 1
    return count

def main():
    max = input("Enter an integer value to see how many numbers less than it contain consecutive significant bits: ")
    num = int(max)
    count = decrement(num)

    print("The total amount of numbers less than",max,"was",count,".")

if __name__ == '__main__':
    main()

File: EX03_03/C++/BinaryRepPart2/BinaryRepPart2.cpp
// BinaryRepresentation.cpp : Defines the entry point for the console application.


// Yuriy Fedas - Assembly Programming - Fall 2018

#include "stdafx.h"
#include <iostream>

bool check(int num) {

	while (num != 0) {
		if ((num & 3) == 3) {
			return true;
		}
		else {
			num = num >> 1;
		}
	}
	// Default case
	return false;
}

int decrement(int num) {
	num--;
	int count = 0;

	while (num != 0) {
		if (check(num))
			count++;
		num--;
	}

	return count;
}

int main()
{
	int num;
	std::cout << "Enter an integer value to see how many numbers less than it contain consecutive significant bits: " << std::endl;
	std::cin >> num;

	std::cout << "The total amount of integers less than " << num << " was " << decrement(num) << ".\n";
}



